lab_7_llm/main.py

```diff
@@ -41,27 +43,90 @@ def obtain(self) -> None:
         Raises:
             TypeError: In case of downloaded dataset is not pd.DataFrame
         """
+        self._raw_data = load_dataset("HuggingFaceH4/no_robots", split='train_sft').to_pandas()
 
 
 class RawDataPreprocessor(AbstractRawDataPreprocessor):
     """
     A class that analyzes and preprocesses a dataset.
     """
 
+    @report_time
     def analyze(self) -> dict:
         """
         Analyze a dataset.
 
         Returns:
             dict: Dataset key properties
         """
+        properties_dict = {"dataset_number_of_samples": self._raw_data.shape[0],
+                           "dataset_columns": self._raw_data.shape[1],
+                           "dataset_empty_rows": self.search_empty_cells(),
+                           "dataset_duplicates": self.search_duplicates()}
+
+        self._raw_data = self._raw_data.dropna()
+        properties_dict["dataset_sample_min_len"] = min(self.get_min_len("prompt"),
+                                                        self.get_min_len("messages"))
+        properties_dict["dataset_sample_max_len"] = max(self.get_max_len("prompt"),
+                                                        self.get_max_len("messages"))
+        self.search_duplicates()
+        return properties_dict
+
+    ## I'm so done with this thing, but, unlike, .isna(), it works >_<
+    def search_empty_cells(self):
```

Comment:
I would be grateful if you'd tell me which functions I could use. I tried .isnull() and .isna() for searching. I failed to find what this symbol is. I also tried to replace zero-length strings with Nan, but it didn't work either. As for .duplicated() function, it raises an error because of NumPy array.